src/lib/views/account-settings/DeleteAccount.svelte

```diff
@@ -7,28 +7,98 @@
   import type { AppStore } from "$lib/stores/types";
   import Button from "../../../lib/components/Button.svelte";
   import isMounted from "../../../lib/is-mounted";
+
   const dispatch = createEventDispatcher();
   const store: AppStore = getContext("rally:store");
+  const errorClass = "mzp-c-field-control mzp-c-field-control--error";
+  const inputClass = "mzp-c-field-control";
+
+  let btnDisabled = true;
+  let inputPasswordClass;
+  let inputItemsVisible = false;
+  let passwordEl;
+  let errText = null;
+  let passwordVisible = false;
+  let fireBaseErr = null;
 
   const offboardURL = "https://rally.mozilla.org/account-deleted";
   const mounted = isMounted();
 
   let leaveModal = false;
+  let isGoogleAccount;
   let Dialog;
 
   onMount(async () => {
+    inputPasswordClass = inputClass;
+    const userProvider = await store.getUserProvider();
+    isGoogleAccount =
+      userProvider &&
+      userProvider.length &&
+      userProvider[0].providerId &&
+      userProvider[0].providerId === "google.com";
+    btnDisabled = !isGoogleAccount;
     Dialog = (await import("../../../lib/components/Dialog.svelte")).default;
   });
 
+  const handleChange = (e) => {
+    const name = e.srcElement.name;
+    inputPasswordClass = inputClass;
+    errText = null;
+    btnDisabled = false;
+    if (name === "id_user_pw") {
+      inputItemsVisible = true;
+    }
+    if (passwordEl) {
+      if (passwordEl.value === "") {
+        btnDisabled = true;
+      }
+    }
+  };
+
+  const handleToggle = () => {
+    passwordVisible = !passwordVisible;
+  };
+
   const handleSelect = (type) => {
     dispatch("type", {
       text: type,
     });
   };
 
+  const handleNextState = () => {
+    /* if the input fields are not empty, check for firebase errors. */
+    fireBaseErr = localStorage.getItem("authErr");
+    if (fireBaseErr) {
+      setMessage();
+    } else {
+      leaveModal = false;
+      window.location.href = offboardURL;
+    }
+  };
+
+  const setMessage = () => {
+    const wrongPW = "auth/wrong-password";
+    const wrongUser = "auth/user-mismatch";
+    const popupBlocked = "auth/popup-blocked";
+    if (fireBaseErr.indexOf(wrongPW) > -1) {
```

Comment:
I think you could do `firebaseErr.includes(wrongPw)` if you want, I see we use this pattern elsewhere though so maybe we could replace this holistically later. IIRC `includes` is relatively new, ES2016 or something, and `indexOf` was the best way in ES5.

Reply:
yes I was following the pattern. Generally have been taking the "we'll fix it when we move to React!" approach but maybe that's a bit blasé...